app/objects/c_link.py

```diff
@@ -46,9 +47,8 @@ def states(self):
     @property
     def output(self):
         try:
-            with open('data/results/%s' % self.unique, 'r') as fle:
-                return fle.read()
-        except Exception:
+            return BaseService.get_service('file_svc').read_result_file(self.unique)
```

Comment:
overall, decent stuff. this is the only line I do not like. I don't want objects touching services. let's revise to not require that.

Reply:
I updated things so that results /  `output` get stored on links as an instance attribute instead of getting loaded from a file.  Though this sort of makes me wonder--if we do this, do we really need the results directory anymore since the link output gets written to the object_store?